src/cpu/testers/dr_trace_player/trace_player.cc

```diff
@@ -167,24 +165,60 @@ DRTracePlayer::executeMemInst(DRTraceReader::TraceRef &mem_ref)
 bool
 DRTracePlayer::trySendMemRef(DRTraceReader::TraceRef &mem_ref)
 {
-    PacketPtr pkt = getPacket(mem_ref);
+    auto [pkt, split_pkt] = getPacket(mem_ref);
+
+    if (!retrySplitPkt) {
+        // we should not be sending the first pkt again if
+        // we are retrying only on the second part of the
+        // previously stalled request
```

Comment:
This comment is for the other side of the branch, not this side.

Reply:
If I have understood correctly, the place of the comment should be changed?

Reply:
Correct.

Reply:
Added/fixed comments to hopefully make things better and not worse